sprite.go

```diff
@@ -362,12 +416,36 @@ func (p *Sprite) OnCloned__1(onCloned func()) {
 	})
 }
 
+func (p *Sprite) fireTouchBegin(obj *Sprite) {
+	if p.hasOnTouchBegin {
+		p.doWhenTouchBegin(p, obj)
+	}
+}
+
 func (p *Sprite) fireTouched(obj *Sprite) {
 	if p.hasOnTouched {
 		p.doWhenTouched(p, obj)
 	}
 }
 
+func (p *Sprite) fireTouchEnd(obj *Sprite) {
+	if p.hasOnTouchEnd {
+		p.doWhenTouchEnd(p, obj)
+	}
+}
+
+func (p *Sprite) OnTouchBegin(onTouchBegin func(obj *Sprite)) {
```

Comment:
We should overload `OnTouchBegin` & `OnTouchEnd` to support differen usages just like `OnTouched`.

Reply:
The touch lifecycle interfaces of several sprites are removed.

Reply:
@qlli Different usages of `OnTouchBegin` & `OnTouchEnd` are still unsupported.